sdk/template/template/package.json

```diff
@@ -123,7 +123,10 @@
     "source-map-support": "^0.5.9",
     "typedoc": "0.15.2",
     "typescript": "~4.2.0",
-    "util": "^0.12.1"
+    "util": "^0.12.1",
+    "@azure-tools/test-recorder-new": "~1.0.0",
+    "@azure/storage-queue": "~12.8.0-beta.2",
+    "@azure/data-tables": "~13.0.1"
```

Comment:
@timovv could you please explain in the PR description why the template package is running tests for tables?

Reply:
Added, but to follow up here -- it's so there are examples of tests available for both core v1 and core v2 SDKs (the approach to setting up the recorder differs slightly in each case).

Reply:
I see. Is not the template package already using core v2?

Reply:
It's using core v1 (there's an issue for open for upgrading it: https://github.com/Azure/azure-sdk-for-js/issues/18044).

Reply:
That said, in that way the storage-queue test is somewhat redundant (since it's an example using core v1) but I think there's still value in having it since it shows a more complex use of the sanitizer functionality.

Reply:
I would argue for upgrading template to corev2 as a prerequisite to moving it to the shared recorder. We don't want to do anything "artificial" in the template package that a real developer working in our repo wouldn't actually do. A dev should be able to easily copy it and use it as a starting point.

Reply:
> A dev should be able to easily copy it and use it as a starting point.

If that's the main purpose of the template project, then agreed, we should dump the tests which don't actually go against the template project, although it would be good to make them available somewhere.

Regarding upgrading to core v2, the required changes to the new recorder tests shouldn't be that substantial when we do get around to it. I'm of the opinion that it's out of scope for this particular PR, but it would be good to get that updated sooner rather than later.